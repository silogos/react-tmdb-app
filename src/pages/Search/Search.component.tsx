import { ChevronLeftIcon } from "@heroicons/react/24/outline";
import { NavLink } from "react-router";
import SearchInput from "@/components/SearchInput";

export function Header() {
  return (
    <header className="overflow-hidden bg-gray-300 bg-opacity-20">
      <div className="mx-auto max-w-screen-xl px-8 py-4">
        <div className="flex items-stretch">
          <NavLink
            to="/"
            className="cursor-pointer hover:underline flex flex-row items-center text-base rounded-lg text-nowrap text-gray-400"
          >
            <ChevronLeftIcon className="size-6 inline" />
            <span className="hidden lg:block">Back to list</span>
          </NavLink>
        </div>
      </div>
    </header>
  );
}

export function Search({
  queryValue,
  handleSearch,
}: {
  queryValue: string;
  handleSearch: (val: string) => void;
}) {
  return (
    <>
      <h3 className="text-gray-300 mb-4">Result for: {queryValue}</h3>
      <SearchInput className="mb-4" onSubmit={handleSearch} />
    </>
  );
}
