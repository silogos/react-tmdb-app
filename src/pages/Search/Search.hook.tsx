import { useCallback, useEffect } from "react";
import { useSearchParams } from "react-router";

import useMovies from "@/hooks/useMovies";

function useSearchPage() {
  const [searchParams, setSearchParams] = useSearchParams();
  const queryValue = searchParams.get("query") || "";
  const {
    data,
    isLoading,
    isError,
    hasMoreData,
    fetchNextData,
    setSearchText,
  } = useMovies({
    path: "/search/movie",
    initialSearchText: queryValue,
  });

  const handleSearch = useCallback(
    (searchText: string) => {
      setSearchText(searchText);
      searchParams.set("query", searchText);
      setSearchParams(searchParams);
    },
    [searchParams, setSearchParams, setSearchText]
  );

  useEffect(() => {
    setSearchText(queryValue);
  }, [queryValue]);

  return {
    queryValue,
    data,
    isLoading,
    isError,
    hasMoreData,
    fetchNextData,
    handleSearch,
  };
}

export default useSearchPage;
