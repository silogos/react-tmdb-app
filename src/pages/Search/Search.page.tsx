import PageLayout from "@/layouts/Page";
import MovieListComponent from "@/components/MovieList";

import { Header, Search } from "./Search.component";
import useSearchPage from "./Search.hook";

function SearchPage() {
  const {
    queryValue,
    fetchData,
    data,
    isLoading,
    isError,
    hasMoreData,
    fetchNextData,
    handleSearch,
  } = useSearchPage();

  return (
    <PageLayout
      title={`Results for "${queryValue}"`}
      titleAppendAppName={true}
      backgroundImagePath={data[0]?.backdrop_path}
    >
      <div className="relative w-full min-h-screen">
        <Header />

        <section id="movie-list" className="overflow-hidden py-4">
          <div className="mx-auto max-w-screen-xl px-6 lg:px-8">
            <Search queryValue={queryValue} handleSearch={handleSearch} />
            <MovieListComponent
              movies={data}
              isLoading={isLoading}
              isError={isError}
              onReload={fetchData}
              hasMoreData={hasMoreData}
              onLoadMore={fetchNextData}
            />
          </div>
        </section>
      </div>
    </PageLayout>
  );
}

export default SearchPage;
