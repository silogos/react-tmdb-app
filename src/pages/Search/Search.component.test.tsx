import { describe, it, expect, vi } from "vitest";
import { Header, Search } from "./Search.component";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router";

vi.mock("@/components/SearchInput");

describe("Search Page Component", () => {
  it("renders Header correctly", () => {
    const component = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(component.container).toMatchSnapshot();
  });
  it("renders Search correctly", () => {
    const component = render(<Search queryValue="" handleSearch={() => {}} />);

    expect(component.container).toMatchSnapshot();
  });
});
