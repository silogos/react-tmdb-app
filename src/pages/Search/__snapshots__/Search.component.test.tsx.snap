// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Page Component > renders Header correctly 1`] = `
<div>
  <header
    class="overflow-hidden bg-gray-300 bg-opacity-20"
  >
    <div
      class="mx-auto max-w-screen-xl px-8 py-4"
    >
      <div
        class="flex items-stretch"
      >
        <a
          aria-current="page"
          class="cursor-pointer hover:underline flex flex-row items-center text-base rounded-lg text-nowrap text-gray-400 active"
          data-discover="true"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="size-6 inline"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5 8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="hidden lg:block"
          >
            Back to list
          </span>
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Search Page Component > renders Search correctly 1`] = `
<div>
  <h3
    class="text-gray-300 mb-4"
  >
    Result for: 
  </h3>
  <div
    data-testid="SearchInput"
  />
</div>
`;
